### PR TITLE
refactor: disambiguate method names and overloads

### PR DESCRIPTION
**Pull request by FrankChen021**
Created by the GPT-5.6-Sol model.

## Summary

- enable PMD's `MissingOverride` rule in the normal Maven static-check lifecycle
- add a PMD rule that rejects same-class method names differing only by capitalization, while excluding inherited API overrides
- rename the conflicting `HttpRemoteTaskRunner` worker-info accessor caught by that rule
- give safe private and test-only overloads semantic names so callers cannot silently bind to a more-specific overload
- clarify the cache configuration test names that previously differed only by capitalization

Most missing-override findings in the broader CodeQL pattern are generated sources. The only handwritten `IntSet.clone()` finding is not a valid Java override and cannot accept `@Override`. Required Servlet/interface overloads and compatibility APIs are intentionally preserved.

## Verification

- `mvn -ntp -B pmd:check -Dweb.console.skip=true -DskipTests -T1C`
- `mvn -ntp -B checkstyle:check -pl processing,server,indexing-service,extensions-core/kafka-indexing-service -Dweb.console.skip=true -DskipTests`
- `mvn -ntp -B test-compile -pl processing,server,indexing-service,extensions-core/kafka-indexing-service -am -Dweb.console.skip=true -DskipTests -T1C`
- 68 focused tests across parser, nested expressions, group-by, sorting, cache configuration, and the affected Kafka supervisor paths

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/expression/NestedDataExpressions.java [hunk 1]
```diff
@@ -854,7 +854,7 @@ public ExpressionType getOutputType(InputBindingInspector inspector)
   }
 
   @Nullable
-  static Object unwrap(ExprEval input)
+  static Object unwrapEval(ExprEval input)
   {
     return unwrap(input.value());
   }
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply processing/src/main/java/org/apache/druid/query/expression/NestedDataExpressions.java:860]: Fixed in 54968c1a91: unwrapEval now accepts ExprEval<?>, removing the raw type while preserving behavior. NestedDataExpressionsTest passes (12 tests), and Maven validate passes with Checkstyle, PMD, and Enforcer.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java [hunk 2]
```diff
@@ -1385,7 +1385,7 @@ public List<String> getBlacklistedWorkers()
     ).collect(Collectors.toList());
   }
 
-  public Collection<ImmutableWorkerInfo> getBlackListedWorkers()
+  public Collection<ImmutableWorkerInfo> getBlacklistedWorkerInfos()
```
Comment (FrankChen021) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java:1398]: P2 Preserve the existing public getBlackListedWorkers API

This removes the public `getBlackListedWorkers()` method and replaces it with `getBlacklistedWorkerInfos()`. Downstream extensions or tests may fail to compile, and existing binaries may hit `NoSuchMethodError`. Keep the old method as a deprecated delegating alias or explicitly document this incompatible API change.

**Review by FrankChen021 — COMMENTED** [review]
| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 0 |
| P2 | 1 |
| P3 | 0 |
| Total | 1 |

Reviewed 9 of 9 changed files, including surrounding callers and tests. Found one P2 public API compatibility issue; the NestedDataExpressions rename is behavior-preserving.

---

This is an automated review by Codex GPT-5.6-Sol

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java (hunk 2 above)
Comment (FrankChen021) [reply indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java:1398]: Fixed in `afbbe686eb` and pushed. Restored `getBlackListedWorkers()` as a deprecated delegating alias to `getBlacklistedWorkerInfos()`, preserving source and binary compatibility while retaining the new semantic name. Focused `HttpRemoteTaskRunnerTest` passed (21 tests), and Maven Checkstyle/PMD validation passed. Reviewed 9 of 9 changed files.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the updated code for correctness, edge cases, API compatibility, and integration risks; no issues found. The deprecated API wrapper is restored, resolving the previous compatibility concern.

Reviewed 9 of 9 changed files.

---

This is an automated review by Codex GPT-5.6-Sol

**Review by GWphua — APPROVED** [review]
